modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/http/routes/NodeParametersRoutes.scala

```diff
@@ -52,6 +58,62 @@ final case class NodeParametersRoutes[F[_]: Async](
       maybeSnapshot <- getLatestFullSnapshot
     } yield maybeSnapshot.flatMap(_.value.info.updateNodeParameters.flatMap(_.get(nodeId)))
 
+  @derive(eqv, show, encoder)
+  case class NodeParametersInfo(
+    node: PeerInfo,
+    delegatedStakeRewardParameters: DelegatedStakeRewardParameters,
+    nodeMetadataParameters: NodeMetadataParameters
+  )
+
+  object SortOrder extends Enumeration {
+    type SortOrder = Value
+    val Asc, Desc = Value
+
+    def fromRequest(req: Request[F]): Either[String, SortOrder] =
+      req.params.get("sortOrder") match {
+        case None => Right(Asc)
+        case Some(s) =>
+          val res = SortOrder.values.find(_.toString.toLowerCase == s.toLowerCase)
+          Either.cond(res.isDefined, res.get, "Unsupported sort order")
+      }
```

Comment:
take a look at http4s documentation, there is a section for handling query params and path params. instead of manually looking for a param and handling it, you can put it into the url when creating a route.